lib/parser.js

```diff
@@ -42,6 +44,10 @@ async function parse(asyncapiYAMLorJSON, options = {}) {
   let initialFormat;
 
   options.path = options.path || `${process.cwd()}${path.sep}`;
+  //because of Ajv lacks support for circular refs, parser should not resolve them before Ajv validation and first needs to ignore them and leave circular $refs to successfully validate the document
+  //this is done pair to advice from Ajv creator https://github.com/ajv-validator/ajv/issues/1122#issuecomment-559378449
+  //later we perform full dereference of circular refs if they occure
+  options.dereference = { circular: 'ignore'};
```

Comment:
I think it's safe to say now that we can remove `dereference` from the `options` object. I mean not allowing the user to customize its value because we need it ourselves and we're not using the passed value anyway.

Reply:
good point, done